app/scripts/directives/processTemplateDialog.js

```diff
@@ -124,5 +234,78 @@
     function instantiateTemplate() {
       $scope.$broadcast('instantiateTemplate');
     }
+
+    function filterForKeywords(searchText, items) {
+      return KeywordService.filterForKeywords(items, ['name', 'tags'], KeywordService.generateKeywords(searchText));
+    }
+
+    function filterChange(filters) {
+      // only use filters which have a filter criteria 'value'
+      // prevents applying an empty keyword filter
+      // TODO: can remove the following line of code after angular-patternfly issue is fixed:
+      //   https://github.com/patternfly/angular-patternfly/issues/509
+      filters = _.filter(filters, 'value');
+
+      ctrl.filterConfig.appliedFilters = filters;
+      filterItems();
+    }
+
+    function filterItems() {
+      ctrl.filteredItems = ctrl.catalogItems;
+      if (ctrl.filterConfig.appliedFilters && ctrl.filterConfig.appliedFilters.length > 0) {
+        _.each(ctrl.filterConfig.appliedFilters, function(filter) {
+          ctrl.filteredItems = filterForKeywords(filter.value, ctrl.filteredItems);
+        });
+      }
+
+      // Deselect the currently selected template if it was filtered out
+      if (!_.includes(ctrl.filteredItems, ctrl.selectedTemplate)) {
+        ctrl.templateSelected();
+      }
+
+      updateFilterControls();
+    }
+
+    function updateFilterControls() {
+      if (ctrl.filterConfig.appliedFilters.length > 0) {
+        ctrl.filterConfig.resultsCount = ctrl.filteredItems.length;
+        $('.toolbar-pf-results h5').text(ctrl.filterConfig.resultsCount + ' of ' + ctrl.totalCount + ' items');
+      } else {
+        $('.toolbar-pf-results h5').text(ctrl.totalCount + (ctrl.totalCount === 1 ? ' item' : ' items'));
+        if (ctrl.totalCount <= 1) {
+          $('.filter-pf.filter-fields input').attr('disabled', '');
+        } else {
+          $('.filter-pf.filter-fields input').removeAttr("disabled");
+        }
+      }
+    }
+
+    var updateProjects = function() {
+      var filteredProjects = _.reject(ctrl.unfilteredProjects, 'metadata.deletionTimestamp');
+      var projects = _.sortBy(filteredProjects, $filter('displayName'));
+      ctrl.searchEnabled = !_.isEmpty(filteredProjects);
+
+      ctrl.templateProjects = RecentlyViewedProjectsService.orderByMostRecentlyViewed(projects);
+    };
+
+    var updateProjectData = function(projectData) {
+      ctrl.unfilteredProjects = _.toArray(projectData.by("metadata.name"));
+      updateProjects();
+    };
+
+    function listProjects() {
+      if (!projectListWatcher) {
+        ProjectsService.list().then(function(projectData) {
+          updateProjectData(projectData);
+
+          if (!ProjectsService.isProjectListIncomplete() && _.size(ctrl.unfilteredProjects) <= MAX_PROJECTS_TO_WATCH) {
+            projectListWatcher = ProjectsService.watch($scope, updateProjectData);
```

Comment:
I'm not sure it's worth a watch here. I'd probably just list projects to populate the dropdown. This is consistent with what we do in other project selects in the console.

Reply:
Removed the watch.